### ISSUE TITLE
Add Save() and Load() to NavigationService.

### ISSUE THREAD

**Issue by JerryNixon** [Windows-XAML/Template10#1222]
Add Save() and Load() to NavigationService. This is already handed externally, but it should be exposed so developers can handle it manually. This might require an additional property like AutoSaveLoadState.

**Comment by callummoffat** [comment]
I'm already doing work to fix #1101. I'll work on this while I'm at it. What do you think needs to be done?

**Comment by callummoffat** [comment]
Let me clarify: I'm already doing work in the NavigationService to resolve #1101. I'll work on adding this while I'm at it. 

Do you have any guidelines for how you want it implemented, Jerry?

**Comment by JerryNixon** [comment]
Yes, 

Here are some ongoing rules:
1. Don't break the current signatures. 
2. Update the interfaces 
3. Make everything async 

Also, when we do the rewrite, I plan on following the following pattern for public methods:

``` csharp
private Func<Task> _saveAsync;
public Func<Task> SaveAsync
{
    get { return _saveAsync ?? DefaultSaveImplementationAsync; }
    set { _saveAsync = value; }
}
private async Task DefaultSaveImplementationAsync()
{
    await Task.CompletedTask;
}

private Func<Task> _loadAsync;
public Func<Task> LoadAsync
{
    get { return _loadAsync ?? DefaultLoadImplementationAsync; }
    set { _loadAsync = value; }
}
private async Task DefaultLoadImplementationAsync()
{
    await Task.CompletedTask;
}
```

Make sense? It still interacts like a method to the developer, but also allows them to update the implementation without having to inherit from the original class. I only wish I had built BootStrapper like this back in the day. I think I will be able to update most of the T10 classes over time with this pattern without any code change required for developers already using T10. Anyway, I recommend you go ahead and do it this way. I am not sure if this pattern has a name or if I invented it in my head. Let's refer to it as the assignable method pattern until we learn otherwise that someone dreamed this up 20 years ago and everybody who is anybody is already using it! :) But you get the idea, right?

**Comment by JerryNixon** [comment]
I also think it's possible that you could use some possible, optional parameters. Not sure optional parameters would be supported with this new Jerry pattern, unfortunately. Dang. I will need to keep thinking about how to handle those. At least we're trying!

**Comment by callummoffat** [comment]
I can do that. I'll work on adding this functionality using the "Nixon method". Like the name?

**Comment by callummoffat** [comment]
After that, I'll start testing the process for migrating stuff over.

**Comment by callummoffat** [comment]
This commit is resolved in PR #1225
